### PR TITLE
Add generator

### PR DESCRIPTION
**Pull request by jansepar**
Status: **Opened for visibility**
Linked PRs: https://github.com/mobify/progressive-web-sdk/pull/64, https://github.com/mobify/progressive-web-scaffold/pull/30
## Changes
- Adds generator script
## How to test-drive this PR
- Follow the steps in the README to generate a project
- Ensure it works on Windows
## TODOs
- [ ] Change back to point to downloading develop

### CODE REVIEW

File: generate.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+fi
+
+# Prepare new project directory
+project_dir="$MYDIR/$project_slug"
+echo "Setting up new project in $project_dir"
+mkdir "$project_dir"
+cd "$project_dir" || exit
+
+printf "\nDownloading Progressive Web project scaffold\n"
+WORKING_DIR=$(mktemp -d /tmp/progressive-web-scaffold.XXXXX)
```
Comment (DanielleB-R) [inline generate.sh:38]: Do you know if the git-bash has a `/tmp` directory?

File: generate.sh [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+trap 'rm -rf "$WORKING_DIR"' EXIT
+curl --progress-bar -L "$SCAFFOLD_URL" -o "$WORKING_DIR/progressive-web-scaffold-$SCAFFOLD_VERSION_OR_BRANCH.zip"
+cd "$WORKING_DIR" || exit
+unzip -q "$WORKING_DIR/progressive-web-scaffold-$SCAFFOLD_VERSION_OR_BRANCH.zip"
```
Comment (DanielleB-R) [inline generate.sh:42]: Or unzip?

File: generate.sh (hunk 1 above)
Comment (jansepar) [inline generate.sh:38]: Yep, seemed to, the files successfully downloaded there.

**Comment by DanielleB-R** [comment]
Seems alright, though it's beyond my Bash skills.

File: generate.sh (hunk 2 above)
Comment (jansepar) [inline generate.sh:42]: Sorry not sure what this is suggesting, are you suggesting `unzip` without the `-q`?

Comment (DanielleB-R) [inline generate.sh:42]: No, I was wondering whether unzip was available on Windows/git-bash. But if you've run it there then I have no concerns.

File: generate.sh [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+# Get the project slug from the user
+read -p'--> What is your project slug? (should match slug on Mobify Cloud) ' project_slug
+
+# Get the project url from the user
+read -p'--> What is the project url? ' project_url
+
+# $project_slug must not contain special characters.
+project_slug=$(echo "$project_slug" | tr -dc '[:alnum:]\n\r' | tr '[:upper:]' '[:lower:]' | tr -d ' ')
+
+read -p"--> Continue with the project name '$project_slug'? (y/n) " -n 1 -r
```
Comment (mikenikles) [inline generate.sh]: As discussed, let's remove that question and line 32. Dashes are ok for progressive web project directories.

**Comment by mikenikles** [comment]
LGTM :+1: once https://github.com/mobify/generator-progressive-web/pull/1/files#r74112189 has been addressed.

**Pull request merged**